drivers/xen/hypervisor.c

```diff
@@ -81,11 +58,34 @@ static const char *param_name(int op)
 	return names[op];
 }
 
+/**
+ * hvm_get_parameter_maintain_dcache - function to obtain a HVM
+ * parameter value.
+ * @idx: HVM parameter index
+ * @value: Value to fill in
+ *
+ * There is no guarantee that cache is going to be clean when
+ * writing xhv. So we need to use invalidate_dcache_range()
+ * before writing it.
+ * In other words, if we start writing in xhv when the cache
+ * flashes, data will be overwritten with old values.
+ *
+ * Example of usage:
+ * In Xen for ARM, there is such a rule that pages must have a
+ * certain attribute as they seen by the vCPU and Xen. Before MMU
+ * setup, we work with the cache turned off, but Xen expects that
```

Comment:
which cache

Reply:
Xen doesn't care about the caches at all. what it expects is that page attributes are equally set for all the entities working with that page. Thus, it defines its expectations in the ABI which allows avoiding manual data cache maintenance